juju/charm.py

```diff
@@ -13,7 +13,7 @@ class LeaderElected(EventBase): pass
 class LeaderSettingsChanged(EventBase): pass
 
 
-class CharmEvents(EventsBase):
+class CharmEventsBase(EventsBase):
```

Comment:
That feels a bit dubious. Charm has no role outside of being a base, and has a name conflict that we needed to solve per our discussion. This is the actual charm events, and doesn't need to be subclasses to be useful, as we can clearly see below.

Can we keep this as CharmEvents, or you see a reason not to?

Reply:
The argument about CharmEvents not needing subclasses is good, especially if we extend it to support endpoint-specific `Event`s based on metadata.yaml.